snapcraft/elf/_elf_file.py

```diff
@@ -0,0 +1,520 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2016-2022 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Helpers to parse and handle ELF binary files."""
+
+import contextlib
+import os
+import re
+import subprocess
+from pathlib import Path
+from typing import Dict, List, Optional, Set, Tuple, Union
+
+import elftools.common.exceptions
+import elftools.elf.constants
+import elftools.elf.dynamic
+import elftools.elf.elffile
+import elftools.elf.gnuversions
+import elftools.elf.sections
+import elftools.elf.segments
+from craft_cli import emit
+from elftools.construct import ConstructError
+from pkg_resources import parse_version
+
+from snapcraft import utils
+
+from . import errors
+
+_ElfArchitectureTuple = Tuple[str, str, str]
+_SonameCacheDict = Dict[Tuple[_ElfArchitectureTuple, str], Path]
+
+_DEBUG_INFO: Union[str, bytes] = ".debug_info"
+_DYNAMIC: Union[str, bytes] = ".dynamic"
+_GNU_VERSION_D: Union[str, bytes] = ".gnu.version_d"
+_GNU_VERSION_R: Union[str, bytes] = ".gnu.version_r"
+_INTERP: Union[str, bytes] = ".interp"
```

Comment:
We shouldn't need the Union anymore, this is from the deb days

Reply:
Removed and kept as strings.